Edge/edge on kubernetes (#1399)
SoftwareAG/c8y-docs: content/edge-k8s/edge-custom-resource-definition-bundle/k8-edge-mongodb.md

---
weight: 36
title: MongoDB
layout: redirect
---

This field is necessary for one or more of the following reasons:

* To specify the MongoDB admin credentials.
* To configure an externally hosted MongoDB server.

> * To specify resource limits for the MongoDB server containers deployed by the Operator.

Reviewer: ```suggestion
* To specify resource limits for the MongoDB server containers deployed by the operator.
```

Author: the word Operator is intentionally kept in uppercase.

Reviewer: then use the full name "Edge Operator". otherwise it should be lower case

Author: Sure. Will update accordingly.

|<div style="width:170px">Field</div>|Required|<div style="width:115px">Type</div>|Default|Description|
|:---|:---|:---|:---|:---|

> |credentialsSecretName|No|String|Defaults to **`databaseAdmin`** and **`admin-pass`** as the database admin user and password for the Operator managed MongoDB or fails with validation error for the externally hosted MongoDB server.<br><br>{{< c8y-admon-info >}}For the MongoDB managed by the Operator, it is recommended to provide the database admin credentials secret, rather than relying on the default credentials assigned by the Operator.{{< /c8y-admon-info >}}|Name of the Kubernetes Secret containing the database admin credentials with which the Operator managed MongoDB must be configured or the database admin credentials of the externally hosted MongoDB server. For more information, see [MongoDB Credentials Secret](/edge-k8s/edge-custom-resource-definition/#k8-edge-mongodb-cred-secret). <p>{{< c8y-admon-info >}}The Operator retrieves this secret from the **`EDGE-CR-NAMESPACE`**. Ensure that this secret is created before initiating the {{< product-c8y-iot >}} Edge deployment or update process.{{< /c8y-admon-info >}}

Reviewer: ```suggestion
|credentialsSecretName|No|String|Defaults to **`databaseAdmin`** and **`admin-pass`** as the database admin user and password for the operator-managed MongoDB or fails with a validation error for the externally hosted MongoDB server.<br><br>For the MongoDB managed by the operator, it is recommended to provide the database admin credentials secret, rather than relying on the default credentials assigned by the operator.|Name of the Kubernetes secret containing the database admin credentials with which the operator-managed MongoDB must be configured or the database admin credentials of the externally hosted MongoDB server. For more information, see [MongoDB credentials secret](/edge-k8s/edge-custom-resource-definition/#k8-edge-mongodb-cred-secret). The Operator retrieves this secret from the `EDGE-CR-NAMESPACE`. Ensure that this secret is created before initiating the {{< product-c8y-iot >}} Edge deployment or update process.
```

Author: The requirement is to have it as Info in the table.

> |connectionString|No|String|If not provided, the Operator installs a single node MongoDB server and configures it with the admin credentials provided in `spec.mongodb.credentialsSecretName`|Connection string of the externally hosted MongoDB server. URI Format: `mongodb://host1[:port1][,...hostN[:portN]]`<br>{{< c8y-admon-info >}}If you do not provide this value, the Operator installs a single node MongoDB server. Once {{< product-c8y-iot >}} Edge is installed and configured to use the Operator managed MongoDB, you cannot provide the `connectionString` to use an externally hosted MongoDB.{{< /c8y-admon-info >}}

Reviewer: ```suggestion
|connectionString|No|String|If not provided, the operator installs a single node MongoDB server and configures it with the admin credentials provided in `spec.mongodb.credentialsSecretName`|Connection string of the externally hosted MongoDB server. URI format: `mongodb://host1[:port1][,...hostN[:portN]]`<br>If you do not provide this value, the operator installs a single node MongoDB server. Once {{< product-c8y-iot >}} Edge is installed and configured to use the operator-managed MongoDB, you cannot provide the `connectionString` to use an externally hosted MongoDB.
```

Author: Done.

> |tlsSecretName|No|String||Secret for supplying the Certificate Authority (CA) certificate to trust. For more information, [External hosted MongoDB TLS Secret](/edge-k8s/edge-custom-resource-definition/#k8-edge-external-hosted-mongodb-tl-secret).<br>{{< c8y-admon-info >}}The Operator retrieves this secret from the **`EDGE-CR-NAMESPACE`**. Ensure that this secret is created before initiating the {{< product-c8y-iot >}} Edge deployment or update process.{{< /c8y-admon-info >}}

Reviewer: ```suggestion
|tlsSecretName|No|String||Secret for supplying the Certificate Authority (CA) certificate to trust. For more information, see [Externally hosted MongoDB TLS secret](/edge-k8s/edge-custom-resource-definition/#k8-edge-external-hosted-mongodb-tl-secret).<br>The operator retrieves this secret from the `EDGE-CR-NAMESPACE`. Ensure that this secret is created before initiating the {{< product-c8y-iot >}} Edge deployment or update process.
```

Author: Done

> |resources.limits|Yes|Structure|Defaults to CPU Limit: 3000m<br>Memory Limit: 6GB|Specify resource limits for the MongoDB server pod. For more information see [Resource Limits Specification](/edge-k8s/edge-custom-resource-definition/#k8-edge-resources-limits-spec).

Reviewer: ```suggestion
|resources.limits|Yes|Structure|Defaults to CPU Limit: 3000m<br>Memory Limit: 6GB|Specify resource limits for the MongoDB server pod. For more information see [Resource limits specification](/edge-k8s/edge-custom-resource-definition/#k8-edge-resources-limits-spec).
```
